pkg/utils/subscription.go

```diff
@@ -1228,6 +1228,42 @@ func IsReadyPlacementDecision(clReader client.Reader) bool {
 	return true
 }
 
+// IsReadySubscription check if Subscription API is ready or not.
+func IsReadySubscription(clReader client.Reader) bool {
+	subList := appv1.SubscriptionList{}
+
+	listopts := &client.ListOptions{}
+
+	err := clReader.List(context.TODO(), &subList, listopts)
+	if err != nil {
+		klog.Error("Subscription API NOT ready: ", err)
+
+		return false
+	}
+
+	subStatusList := appsubReportV1alpha1.SubscriptionStatusList{}
+
+	err = clReader.List(context.TODO(), &subStatusList, listopts)
+	if err != nil {
+		klog.Error("SubscriptionStatus API NOT ready: ", err)
+
+		return false
+	}
+
+	subReportList := appsubReportV1alpha1.SubscriptionReportList{}
```

Comment:
Do we need to check this on the managed cluster?

Reply:
No the managed clusters don't have the subscriptionreport CRD installed.

Reply:
@lennysgarage unless I read it wrong, I think the current PR is checking it on the managed cluster as well.

Reply:
Yup you're correct it is... I will update it

Reply:
@mikeshng  Added a check on line 270 for this to only run on the hub.

Reply:
Sorry @lennysgarage for not being clear but I think we should check on the managed cluster too just the CRDs/APIs that needs to be checked (appsub and appsubstatus).

Reply:
Oh okay I see now, no problem so we only want to check if the appsub and appsubstatus CRDs exist on both the hub and managed clusters.

Reply:
We should check on hub all 3 but managed only the ones that exist.